File: backend/entity/money_donation.go
package entity

import (
	"gorm.io/gorm"
)

type MoneyDonation struct {
	gorm.Model
	Amount          float64 `json:"amount"`
	PaymentType     string  `json:"payment_type"`
	NextPaymentDate string  `json:"next_payment_date"`
	BillingDate     string  `json:"billing_date"`
	TransactionRef  string  `json:"transaction_ref"`
	Status          string  `json:"status"`

	DonationID uint     `json:"donation_id"`
	Donation   Donation `gorm:"foreignKey:DonationID"`

	PaymentMethodID uint           `json:"payment_method_id"`
	PaymentMethod   *PaymentMethod `gorm:"foreignKey:PaymentMethodID"`
}

File: frontend/src/interfaces/MoneyDonation.ts
export interface MoneyDonationInterface {
    money_id?: number;
    amount?: number;
    payment_type?: "one-time" | "monthly";
    next_payment_date?: Date | string;
    billing_date?: string;
    transaction_ref?: string;
    status?: "success" | "complete";
    donation_id?: number;
    payment_method_id?: number;
}